icu4c/source/common/unicode/umachine.h

```diff
@@ -268,50 +268,6 @@
  */
 typedef int8_t UBool;
 
-/**
- * \def U_DEFINE_FALSE_AND_TRUE
- * Normally turns off defining macros FALSE=0 & TRUE=1 in public ICU headers.
- * These obsolete macros sometimes break compilation of other code that
- * defines enum constants or similar with these names.
- * C++ has long defined bool/false/true.
- * C99 also added definitions for these, although as macros; see stdbool.h.
- *
- * You may transitionally define U_DEFINE_FALSE_AND_TRUE=1 if you need time to migrate code.
- *
- * @internal ICU 68
- */
-#ifdef U_DEFINE_FALSE_AND_TRUE
-    // Use the predefined value.
-#elif defined(U_COMBINED_IMPLEMENTATION) || \
-        defined(U_COMMON_IMPLEMENTATION) || defined(U_I18N_IMPLEMENTATION) || \
-        defined(U_IO_IMPLEMENTATION) || defined(U_LAYOUTEX_IMPLEMENTATION) || \
-        defined(U_TOOLUTIL_IMPLEMENTATION)
-    // Inside ICU: Keep FALSE & TRUE available.
-#   define U_DEFINE_FALSE_AND_TRUE 1
-#else
-    // Outside ICU: Avoid collision with non-macro definitions of FALSE & TRUE.
-#   define U_DEFINE_FALSE_AND_TRUE 0
-#endif
```

Comment:
I've now instead simply reverted the public facing code here to what it used to be, so that people using this feature won't experience any change in behaviour at all.